File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        result = 0

        left, right = 0, len(heights) - 1
        while left < right:
            leftHeight, rightHeight = heights[left], heights[right]
            height = min(leftHeight, rightHeight)
            width = right - left
            result = max(result, height * width)
            if leftHeight < rightHeight:
                left += 1
            else:
                right -= 1

        return result

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cs
public class Solution {
    public int FindMin(int[] nums) {
        int left = 0;
        int right = nums.Length - 1;
        int result = nums[0];

        while (left <= right)
        {
            if (nums[left] < nums[right])
            {
                result = Math.Min(result, nums[left]);
                break;
            }
            
            int mid = (left + right) / 2;
            result = Math.Min(result, nums[mid]);
            if (nums[mid] >= nums[left])
            {
                left = mid + 1;
            }
            else
            {
                right = mid - 1;
            }
        }
        return result;

        // while (left <= right)
        // {
        //     int mid = (left + right) / 2;
        //     result = Math.Min(result, nums[mid]);
        //     if (nums[left] <= nums[right])
        //     {
        //         result = Math.Min(result, nums[left]);
        //         break;
        //     }
        //     if (nums[right] < nums[mid])
        //     {
        //         left = mid + 1;
        //     }
        //     else
        //     {
        //         right = mid - 1;
        //     }
        // }
        // return result;
    }
}
